src/sagemaker/image_uris.py

```diff
@@ -117,7 +116,10 @@ def retrieve(
     args = dict(locals())
     for name, val in args.items():
         if is_pipeline_variable(val):
-            raise ValueError("%s should not be a pipeline variable (%s)" % (name, type(val)))
+            raise ValueError(
+                "To retrieve image_uris, the argument %s should not be a pipeline variable (%s)"
```

Comment:
>can we add some description in job class to explain why we need to retrieve the images, and why it can't be a pipeline variable.

Good call out!. You mean we can add a log info in the job classes like:
```
if self.image_uri is None:
  logger.info("image_uri is not presented, using instance_type, <other arg names>, to retrieve image_uri")
```

Then in util, we can modify the error message to be
"To retrieve image_uris, the argument %s should not be a pipeline variable (%s) since pipeline variables are only interpreted in pipeline execution time."

Reply:
LGTM